runtime/vm/OutOfLineINL_jdk_internal_foreign_abi_DowncallLinker.cpp

```diff
@@ -0,0 +1,211 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "VMHelpers.hpp"
+#include "BytecodeAction.hpp"
+#include "UnsafeAPI.hpp"
+#include "j9vmnls.h"
+#include "OutOfLineINL.hpp"
+#include "LayoutFFITypeHelpers.hpp"
+#include "AtomicSupport.hpp"
+
+extern "C" {
+
+#if JAVA_SPEC_VERSION >= 19
+/* jdk.internal.foreign.abi.DowncallLinker: private static synchronized native void resolveRequiredFields(); */
+VM_BytecodeAction
+OutOfLineINL_jdk_internal_foreign_abi_DowncallLinker_resolveRequiredFields(J9VMThread *currentThread, J9Method *method)
```

Comment:
is this the same as `OutOfLineINL_jdk_internal_foreign_abi_ProgrammableInvoker_resolveRequiredFields` ?

Reply:
See comment at https://github.com/eclipse-openj9/openj9/pull/15083#discussion_r902771084

Reply:
The two files are removed and replaced with `/runtime/vm/OutOfLineINL_openj9_internal_foreign_abi_InternalDowncallHandler.cpp` based on our latest solution to avoid duplicating code in Java17/18 and Java19